### ISSUE TITLE
Gitlab Replication not working

### ISSUE THREAD

**Issue by liangminhua** [goharbor/harbor#12552]
**Expected behavior and actual behavior:**
replicate docker registery of gitlab, but harbor core crash and restart.

**Steps to reproduce the problem:**
Scene One
1. Create [gitlab docker registry](https://registry.gitlab.com)

2. Create replication rule:

    - Source registry: gitlab
    - Source resource filter:
        - name: gitlab-org/build/cng/{gitlab-webservice-ce,gitlab-workhorse-ce}

     Start replicate, harbor core will be crash and restart.

Scene Two
1. But Separate the two rules：
    - Source registry: gitlab
    - Source resource filter:
        - name: gitlab-org/build/cng/gitlab-webservice-ce
    - Source registry: gitlab
    - Source resource filter:
        - name: gitlab-org/build/cng/gitlab-workhorse-ce
    
    It will be work.

**Versions:**

- harbor version: [19.03.12]
- docker engine version: [2.0.1]
- docker-compose version: [1.24.1, build 4667896b]

**Additional context:**

``` log
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xb5cdbc]

goroutine 35857 [running]:
github.com/goharbor/harbor/src/pkg/registry/auth.(*authorizer).Modify(0xc0002f5630, 0xc000f4b600, 0x0, 0x0)
        /harbor/src/pkg/registry/auth/authorizer.go:73 +0x5c
github.com/goharbor/harbor/src/pkg/registry.(*client).do(0xc000767ce0, 0xc000f4b600, 0x0, 0x0, 0x0)
        /harbor/src/pkg/registry/client.go:511 +0x596
github.com/goharbor/harbor/src/pkg/registry.(*client).listTags(0xc000767ce0, 0xc00085ed20, 0x52, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /harbor/src/pkg/registry/client.go:224 +0xd5
github.com/goharbor/harbor/src/pkg/registry.(*client).ListTags(0xc000767ce0, 0xc0009db2c0, 0x29, 0x0, 0xc000a71b02, 0x0, 0x7f119d580318, 0xc0013ad5e8)
        /harbor/src/pkg/registry/client.go:200 +0x147
github.com/goharbor/harbor/src/replication/adapter/native.(*Adapter).listArtifacts(0xc00075f740, 0xc0009db2c0, 0x29, 0xc000984340, 0x1, 0x1, 0x8, 0xc0013ad628, 0xc0013ad708, 0xab5b29, ...)
        /harbor/src/replication/adapter/native/adapter.go:219 +0x63
github.com/goharbor/harbor/src/replication/adapter/native.(*Adapter).FetchArtifacts.func1(0xc0013ad740, 0x1)
        /harbor/src/replication/adapter/native/adapter.go:150 +0x8a
github.com/goharbor/harbor/src/common/utils.(*limitedConcurrentRunner).AddTask.func1(0xc0006d7860, 0xc0008aa8c0)
        /harbor/src/common/utils/passports.go:107 +0xd2
created by github.com/goharbor/harbor/src/common/utils.(*limitedConcurrentRunner).AddTask
        /harbor/src/common/utils/passports.go:93 +0x63
```

**Comment by bitsf** [comment]
similar as #12575

**Comment by ywk253100** [comment]
Dup of #11939 and already fixed, the fix will be available on 2.0.2
